src/script/components/MessagesList/Message/Message.tsx

```diff
@@ -0,0 +1,174 @@
+import cx from 'classnames';
+
+import {ServiceEntity} from 'src/script/integration/ServiceEntity';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {getAllFocusableElements, setElementsTabIndex} from 'Util/focusUtil';
+import {isTabKey} from 'Util/KeyboardUtil';
+
+import {ElementType, MessageDetails} from './ContentMessage/asset/TextMessageRenderer';
+import {MessageWrapper} from './MessageWrapper';
+import {useMessageFocusedTabIndex} from './util';
+
+import type {MessageRepository} from '../../../conversation/MessageRepository';
+import type {Conversation} from '../../../entity/Conversation';
+import type {ContentMessage} from '../../../entity/message/ContentMessage';
+import type {DecryptErrorMessage} from '../../../entity/message/DecryptErrorMessage';
+import type {MemberMessage as MemberMessageEntity} from '../../../entity/message/MemberMessage';
+import {Message as BaseMessage} from '../../../entity/message/Message';
+import type {User} from '../../../entity/User';
+import {TeamState} from '../../../team/TeamState';
+
+export interface MessageActions {
+  onClickAvatar: (user: User | ServiceEntity) => void;
+  onClickCancelRequest: (message: MemberMessageEntity) => void;
+  onClickImage: (message: ContentMessage, event: React.UIEvent) => void;
+  onClickInvitePeople: () => void;
+  onClickReactionDetails: (message: BaseMessage) => void;
+  onClickMessage: (event: MouseEvent | KeyboardEvent, elementType: ElementType, messageDetails: MessageDetails) => void;
+  onClickParticipants: (participants: User[]) => void;
+  onClickDetails: (message: BaseMessage) => void;
+  onClickResetSession: (messageError: DecryptErrorMessage) => void;
+  onClickTimestamp: (messageId: string) => void;
+}
+
+export interface MessageParams extends MessageActions {
+  conversation: Conversation;
+  hasReadReceiptsTurnedOn: boolean;
+  isLastDeliveredMessage: boolean;
+  isSelfTemporaryGuest: boolean;
+  message: BaseMessage;
+  /** whether the message should display the user avatar and user name before the actual content */
+  hideHeader: boolean;
+  messageActions: {
+    deleteMessage: (conversation: Conversation, message: BaseMessage) => void;
+    deleteMessageEveryone: (conversation: Conversation, message: BaseMessage) => void;
+  };
+  messageRepository: MessageRepository;
+  // onVisibilityLost?: () => void;
+  selfId: QualifiedId;
+  shouldShowInvitePeople: boolean;
+  teamState?: TeamState;
+  /** whether the message is being accessed using the keyboard (will then show the focus state of the elements) */
+  isFocused: boolean;
+  /** will visually highlight the message when it's being loaded */
+  isHighlighted: boolean;
+  handleFocus: (id: string) => void;
+  handleArrowKeyDown: (e: React.KeyboardEvent) => void;
+  isMsgElementsFocusable: boolean;
+  setMsgElementsFocusable: (isMsgElementsFocusable: boolean) => void;
+}
+
+export const Message = (props: MessageParams & {scrollTo?: () => void}) => {
```

Comment:
maybe make an interface out of this? 
MessageParams & {scrollTo?: () => void
-> MessageProps 🤔